fix(migrations): handle parameters with initializers in inject migration

crisbeto opened the pull request:
Adds some logic to the `inject()` migration that allows is to handle parameter declarations with initializers. This was omitted initially, because we don't officially support such cases, but it came up internally.

devversion commented on packages/core/schematics/test/inject_migration_spec.ts:
```diff
@@ -1501,6 +1501,153 @@ describe('inject migration', () => {
+      `  private foo: Foo;`,
+      ``,
+      `  constructor() {`,
+      `    this.foo = inject(Foo, { optional: true }) ?? createFoo(this.bar);`,
```
Could `this.foo` not be directly initialized above here?

crisbeto replied:
Good point, I tweaked it to ignore `this` accesses.

crisbeto committed:
fix(migrations): handle parameters with initializers in inject migration

Adds some logic to the `inject()` migration that allows is to handle parameter declarations with initializers. This was omitted initially, because we don't officially support such cases, but it came up internally.

AndrewKushnir commented:
This PR was merged into the repository by commit 18991d3210e635dad6a50cda6088d886980aa58a.

The changes were merged into the following branches: main, 19.0.x
